internal/core/src/segcore/Reduce.cpp

```diff
@@ -193,8 +220,14 @@ ReduceHelper::ReduceSearchResultForOneNQ(int64_t qi,
     }
     pk_set_.clear();
     pairs_.clear();
+    group_by_val_set_.clear();
```

Comment:
there is no reason to keep this map very time?

In fact I would more like to have a separate function to implement groupby but not implement in same functions.